File: Selenium/Project/DirectoryMenu.java
package HRM;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;


public class DirectoryMenu {

	static WebDriver driver;
	static WebDriverWait wait;
	
	
	@BeforeClass
	public void setUp() {
		driver = new ChromeDriver();
		driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
		driver.manage().window().maximize();
		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	}
	
	@Test(priority =1)
	@Parameters({"username", "password"})
	public void login_page(String username, String password) {
		driver.findElement(By.id("txtUsername")).sendKeys(username);
		driver.findElement(By.id("txtPassword")).sendKeys(password);
		driver.findElement(By.id("btnLogin")).click();
		Boolean logged_in = driver.findElement(By.xpath("//*[@id = 'welcome']")).isDisplayed();
		Assert.assertTrue(logged_in);
//		Thread.sleep(3000);
	}
	
	@Test(priority =2)
	public void DirectoryPageTitle() throws InterruptedException {
		
		WebElement directory = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[text()= 'Directory']")));
		wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[text()= 'Directory']")));
		Assert.assertTrue(directory.isDisplayed(), "Directory menu should be displayed");
		Thread.sleep(9000);
		directory.click();
		wait.until(ExpectedConditions.visibilityOfElementLocated(By.tagName("h1")));
		Assert.assertEquals(driver.findElement(By.tagName("h1")).getText(), "Search Directory");		
	}
	
	@AfterClass
	public void TearDown() {
		driver.quit();
	}
}

File: Selenium/Project/DirectoryMenu.py
import time

from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


def directory_page_title(driver,wait):

    directory = wait.until(expected_conditions.visibility_of_element_located((By.XPATH, "//*[text()= 'Directory']")))
    wait.until(expected_conditions.element_to_be_clickable((By.XPATH, "//*[text()= 'Directory']")))
    assert directory.is_displayed()
    directory.click()
    time.sleep(5)
    print(driver.find_element(By.TAG_NAME, "h1").text)

with webdriver.Chrome() as driver:
    driver.get("http://alchemy.hguy.co/orangehrm")
    title = driver.title
    print(title)
    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    driver.find_element(By.ID, "btnLogin").click()
    assert driver.find_element(By.XPATH, "//*[@id = 'welcome']").is_displayed();
    print(driver.find_element(By.TAG_NAME, "h1").text)
    wait = WebDriverWait(driver, 5)
    action = ActionChains(driver)
    directory_page_title(driver, wait)
    driver.close()